Source/WebKit/NetworkProcess/cocoa/NetworkDataTaskCocoa.mm

```diff
@@ -179,9 +179,25 @@ static float toNSURLSessionTaskPriority(WebCore::ResourceLoadPriority priority)
         session->setFirstPartyHostIPAddress(requestURL.host().toString(), ipAddress);
 }
 
-void NetworkDataTaskCocoa::applyCookiePolicyForThirdPartyCNAMECloaking(const WebCore::ResourceRequest& request)
+static NSArray<NSHTTPCookie *> *cookiesByCappingExpiry(NSArray<NSHTTPCookie *> *cookies, Seconds ageCap)
 {
-    if (isTopLevelNavigation() || !shouldApplyCookiePolicyForThirdPartyCNAMECloaking())
+    auto *cappedCookies = [NSMutableArray arrayWithCapacity:cookies.count];
+    for (NSHTTPCookie *cookie in cookies)
+        [cappedCookies addObject:WebCore::NetworkStorageSession::capExpiryOfPersistentCookie(cookie, ageCap)];
+    return cappedCookies;
+}
+
+static bool shouldCapCookieExpiry(const WebCore::IPAddress& remote, const WebCore::IPAddress& firstParty)
```

Comment:
I think this one is too broadly named. Or do you think the parameter types gives it away in the signature?

Reply:
Indeed — the intent was for the parameter types to make the purpose clear, but I can make this more explicit by renaming it to `shouldCapCookieExpiryForThirdPartyIPAddress`.